promql/testdata/functions.test

```diff
@@ -67,31 +246,103 @@ eval instant at 15m changes(x[15m])
 
 clear
 
-# Tests for increase().
-load 5m
+# Tests for increase()/xincrease()/xincrease0()//xrate()/xrate0().
+load 5s
 	http_requests{path="/foo"}	0+10x10
-	http_requests{path="/bar"}	0+10x5 0+10x5
+	http_requests{path="/bar"}	0+10x5 0+10x4
 
 # Tests for increase().
-eval instant at 50m increase(http_requests[50m])
+eval instant at 50s increase(http_requests[50s])
 	{path="/foo"} 100
 	{path="/bar"}  90
 
-eval instant at 50m increase(http_requests[100m])
+eval instant at 50s increase(http_requests[100s])
 	{path="/foo"} 100
 	{path="/bar"}  90
 
+# Tests for xincrease()/xincrease0().
+eval instant at 50s xincrease(http_requests[50s])
+	{path="/foo"} 100
+	{path="/bar"}  90
+
+eval instant at 50s xincrease0(http_requests[50s])
+	{path="/foo"} 100
+	{path="/bar"}  90
+
+eval instant at 50s xincrease(http_requests[5s])
+	{path="/foo"} 10
+	{path="/bar"} 10
+
+eval instant at 50s xincrease0(http_requests[5s])
+	{path="/foo"} 100
+	{path="/bar"} 40
+
+eval instant at 50s xincrease(http_requests[3s])
+	{path="/foo"} 6
+	{path="/bar"} 6
+
+eval instant at 50s xincrease0(http_requests[3s])
+	{path="/foo"} 60
+	{path="/bar"} 24
```

Comment:
Why is xincrease and xincrease0 here returning such different values?

Due to this weird DSL, I don't really follow the test setup...

Reply:
That was a magnification factor of tiny samples. xincrease0 was less tolerant of slop at the edges, so it was picking narrow points and magnifying them. With the change I just made to have it tolerate points that are +/- and entire scrapeInterval, it's now much closer:
```
eval instant at 50s xincrease0(http_requests[3s])
	{path="/foo"} 10
	{path="/bar"} 10
```